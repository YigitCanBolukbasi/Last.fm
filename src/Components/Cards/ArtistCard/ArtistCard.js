import {View, Text, Image, TouchableOpacity} from 'react-native';
import React from 'react';
import styles from './ArtistCard.styles';

const ArtistCard = ({artist, onSelect}) => {
  return (
    <TouchableOpacity onPress={onSelect}>
      <View style={styles.container}>
        <Image style={styles.image} />
        <View style={styles.body_container}>
          <View style={styles.title_container}>
            <Text>Artist {artist.name}</Text>
          </View>
          <View style={styles.inner_container}>
            <Text>{artist.playcount}</Text>
            <Text>{artist.listeners}</Text>
          </View>
        </View>
      </View>
    </TouchableOpacity>
  );
};

export default ArtistCard;
