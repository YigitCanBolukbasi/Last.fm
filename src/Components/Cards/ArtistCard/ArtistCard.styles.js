import {StyleSheet, Dimensions} from 'react-native';

export default StyleSheet.create({
  container: {
    flexDirection: 'row',
    padding: 10,
    borderRadius: 12,
    backgroundColor: 'white',
    zIndex: 1,
    height: Dimensions.get('window').height / 6,
  },
  shadow_container: {
    borderColor: 'black',
    padding: 5,
    elevation: 20,
    marginBottom: 10,
    borderRadius: 12,
    shadowColor: '#000',
  },
  image: {
    overflow: 'visible',
    width: 70,
    height: 70,
    borderRadius: 70,
    marginRight: 10,
    alignSelf: 'center',
  },
  body_container: {
    flex: 1,
    flexDirection: 'row',
    margin: 5,
  },
  title_container: {
    flex: 1,
    margin: 5,
  },
  inner_container: {
    flex: 1,
    margin: 5,
    marginTop: 30,
  },
  title: {
    fontSize: 18,
    fontWeight: '700',
    color: 'black',
    borderBottomColor: 'black',
    borderBottomWidth: 1,
    margin: 10,
  },
  artist: {
    fontSize: 18,
    fontWeight: '700',
    color: 'black',
    margin: 10,
  },
  text: {
    fontSize: 12,
    color: 'black',
    margin: 5,
  },
  container_dark: {
    flexDirection: 'row',
    padding: 10,
    borderRadius: 12,
    backgroundColor: 'gray',
    zIndex: 1,
    height: Dimensions.get('window').height / 6,
  },
  title_dark: {
    fontSize: 18,
    fontWeight: '700',
    color: 'white',
    borderBottomColor: 'white',
    borderBottomWidth: 1,
    margin: 10,
  },
  artist_dark: {
    fontSize: 18,
    fontWeight: '700',
    color: 'white',
    margin: 10,
  },
  text_dark: {
    fontSize: 12,
    color: 'white',
    margin: 5,
  },
});
