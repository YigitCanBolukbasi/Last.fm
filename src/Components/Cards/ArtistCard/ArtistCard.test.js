import React from 'react';
import {fireEvent, render, waitFor} from '@testing-library/react-native';
import ArtistCard from './ArtistCard';
import {ThemeProvider} from '../../../Contexts/ThemeContext';

test('should first', () => {
  const items = {
    image: [
      {
        '#text':
          'https://lastfm.freetls.fastly.net/i/u/34s/2a96cbd8b46e442fc41c2b86b821562f.png',
        size: 'small',
      },
      {
        '#text':
          'https://lastfm.freetls.fastly.net/i/u/64s/2a96cbd8b46e442fc41c2b86b821562f.png',
        size: 'medium',
      },
      {
        '#text':
          'https://lastfm.freetls.fastly.net/i/u/174s/2a96cbd8b46e442fc41c2b86b821562f.png',
        size: 'large',
      },
      {
        '#text':
          'https://lastfm.freetls.fastly.net/i/u/300x300/2a96cbd8b46e442fc41c2b86b821562f.png',
        size: 'extralarge',
      },
      {
        '#text':
          'https://lastfm.freetls.fastly.net/i/u/300x300/2a96cbd8b46e442fc41c2b86b821562f.png',
        size: 'mega',
      },
    ],
    listeners: '5354850',
    mbid: 'b95ce3ff-3d05-4e87-9e01-c97b66af13d4',
    name: 'Eminem',
    playcount: '272269112',
    streamable: '0',
    url: 'https://www.last.fm/music/Eminem',
  };
  const comp = render(
    <ThemeProvider>
      <ArtistCard artist={items} />
    </ThemeProvider>,
  );

  expect(comp).toMatchSnapshot();
});

/* test('should first', () => {
  expect(5 + 6).toBe(11);
}); */
