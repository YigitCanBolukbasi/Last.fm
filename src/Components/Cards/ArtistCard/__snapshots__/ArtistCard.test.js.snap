// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should first 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="artistTestButton"
>
  <View
    style={
      Object {
        "borderColor": "black",
        "borderRadius": 12,
        "elevation": 20,
        "marginBottom": 10,
        "padding": 5,
        "shadowColor": "#000",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
          "borderRadius": 12,
          "flexDirection": "row",
          "height": 222.33333333333334,
          "padding": 10,
          "zIndex": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://lastfm.freetls.fastly.net/i/u/300x300/2a96cbd8b46e442fc41c2b86b821562f.png",
          }
        }
        style={
          Object {
            "alignSelf": "center",
            "borderRadius": 70,
            "height": 70,
            "marginRight": 10,
            "overflow": "visible",
            "width": 70,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "margin": 5,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "margin": 5,
            }
          }
        >
          <Text
            style={
              Object {
                "borderBottomColor": "black",
                "borderBottomWidth": 1,
                "color": "black",
                "fontSize": 18,
                "fontWeight": "700",
                "margin": 10,
              }
            }
          >
            Artist
          </Text>
          <Text
            style={
              Object {
                "color": "black",
                "fontSize": 18,
                "fontWeight": "700",
                "margin": 10,
              }
            }
          >
            Eminem
          </Text>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "margin": 5,
              "marginTop": 30,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "black",
                "fontSize": 12,
                "margin": 5,
              }
            }
          >
            Playcount:
            272269112
          </Text>
          <Text
            style={
              Object {
                "color": "black",
                "fontSize": 12,
                "margin": 5,
              }
            }
          >
            Listeners:
            5354850
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
